### PR TITLE
Some support for non-default tracer amount units.

### PR DESCRIPTION
**Pull request by osae**
Simulation based on non-default tracer units has been activated. 
`TracerConfig` keeps track of unit names and generates associated name tags for summary.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/TracerConfig.cpp [hunk 1]
```diff
@@ -142,4 +160,36 @@ bool TracerConfig::operator==(const TracerConfig& other) const {
...
+                            unit_string += unit_system.name(Opm::UnitSystem::measure::liquid_surface_volume);
+                    }
+                    else {
+                        throw std::runtime_error("Tracer summary kw not recognized: " + tracer_kw);
+                    }
+                }
+                return unit_string;
+            }
+        }
+    }
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/TracerConfig.cpp:191]: I suppose we might switch our approach here to reduce the nesting a bit.  On the other hand, the current code clearly does what it needs to do so I'll hold off on that until a later refactoring.
```C++
using M = UnitSystem::measure;

if (tracer_kw.length() <= 4) {
    return {};
}
const auto tracer_name = tracer_kw.substr(4);
auto tracerPos = std::find_if(this->tracers.begin(), this->tracers.end(),
    [&tracer_name](const auto& tracer)
{
    return tracer.name == tracer_name;
});
if ((tracerPos == this->tracers.end()) || tracerPos->unit_string.empty()) {
    return {};
}

auto unit_string = tracerPos->unit_string;
if (tracer_kw[3] == 'R') {
    const auto rateName = std::string_view(unit_system.name(M::rate));
    unit_string += rateName.substr(rateName.find('/'));
}
else if (tracer_kw[3] == 'C') {
    const auto volume_unit = (tracerPos->phase == Phase::GAS)
        ? M::gas_surface_volume
        : M::liquid_surface_volume;
    unit_string.append('/').append(unit_system.name(volume_unit));
}
else if (tracer_kw[3] != 'T') {
    throw std::runtime_error {
        "Tracer summary keyword '" + tracer_kw + "' is not recognized"
    };
}

return unit_string;
```

**Review by bska — APPROVED** [review]
I think is a good start in a complex area.  I see some potential improvements in this code, but those can be introduced later.  For now I'm happy to merge this to enable more widespread testing of tracers.

**Comment by osae** [comment]
Thank you for reviewing and for useful tips.  Please feel free to incorporate the suggestions whenever it is convenient.

**Pull request merged**